### PR TITLE
Adding new link domains to shady-links

### PR DESCRIPTION
**Pull request by sobregosodd**
Added sites observed in malicious samples 

closes https://github.com/DataDog/guarddog/issues/180

### CODE REVIEW

File: guarddog/analyzer/sourcecode/shady-links.yml [hunk 1]
```diff
@@ -25,6 +25,7 @@ rules:
         - pattern-either:
             # complete domains
             - pattern-regex: (http[s]?:\/\/[^\n\[\/\?#"']*?(bit\.ly|discord\.com|workers\.dev|transfer\.sh|filetransfer\.io|sendspace\.com|appdomain\.cloud|backblazeb2\.com\|paste\.ee|ngrok\.io|termbin\.com|localhost\.run|webhook\.site|oastify\.com|burpcollaborator\.me)\/)
+            - pattern-regex: (http[s]?:\/\/[^\n\[\/\?#"']*?(oast\.(pro|live|site|online|fun|me)|api\.telegram\.org|rentry\.co)\/)
```
Comment (christophetd) [inline guarddog/analyzer/sourcecode/shady-links.yml:28]: IIRC we had api.telegram.org in the past and it triggered quite a lot of noise

Comment (sobregosodd) [reply guarddog/analyzer/sourcecode/shady-links.yml:28]: ok, will run it against the top 1k before merge

Comment (sobregosodd) [reply guarddog/analyzer/sourcecode/shady-links.yml:28]: I just had 1 from in the top 1 pypi and 0 hits in top 1k npm.
(could not find it in the repo history)
I'll proceed

**Review by ikretz — APPROVED** [review]
LGTM

**Pull request merged**